server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -96,7 +99,19 @@ public IngestService(ClusterService clusterService, ThreadPool threadPool,
                 threadPool.getThreadContext(), threadPool::relativeTimeInMillis,
                 (delay, command) -> threadPool.schedule(
                     command, TimeValue.timeValueMillis(delay), ThreadPool.Names.GENERIC
-                ), this
+                ), this, index -> {
+                    IndexService indexService = indicesService.indexService(index);
```

Comment:
I assume from the exception message that the IndexService returns null unless it it local to this node ? (it is not very clear from the code if InicesService is per node or per cluster).

Reply:
Yes, `IndicesServices` only contains a `IndexService` instance for locally allocated indices.